enforcer-rules/src/main/java/com/google/cloud/tools/dependencies/enforcer/LinkageCheckerRule.java

```diff
@@ -117,8 +117,8 @@ public void execute(@Nonnull EnforcerRuleHelper helper) throws EnforcerRuleExcep
       }
 
       String projectType = project.getArtifact().getType();
-      if (dependencySection == DependencySection.DEPENDENCIES && !"jar".equals(projectType)) {
-        // When checking non-BOM project, not interested in non JAR artifact
+      if (dependencySection == DependencySection.DEPENDENCIES && "pom".equals(projectType)) {
```

Comment:
what about all the other types: ear, war, javadocs, etc.?

https://maven.apache.org/ref/3.6.1/maven-core/artifact-handlers.html

Reply:
Added exclusion for when not checking non BOM project: pom, java-source, javadoc.